Sources/PackageMetadata/PackageMetadata.swift

```diff
@@ -0,0 +1,147 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Basics
+import Dispatch
+import PackageCollections
+import PackageModel
+import PackageRegistry
+import SourceControl
+
+import struct Foundation.URL
+import protocol TSCBasic.FileSystem
+import func TSCBasic.withTemporaryDirectory
+
+public struct Package {
+    public let identity: PackageIdentity
+    public let location: String?
+    public let branches: [String]
+    public let versions: [Version]
+    public let readmeURL: URL?
+
+    fileprivate init(identity: PackageIdentity, location: String? = nil, branches: [String] = [], versions: [Version], readmeURL: URL? = nil) {
+        self.identity = identity
+        self.location = location
+        self.branches = branches
+        self.versions = versions
+        self.readmeURL = readmeURL
+    }
+}
+
+public struct PackageSearchClient {
+    private let registryClient: RegistryClient
+    private let fileSystem: FileSystem
+    private let observabilityScope: ObservabilityScope
+
+    public init(
+        registryClient: RegistryClient,
+        fileSystem: FileSystem,
+        observabilityScope: ObservabilityScope
+    ) {
+        self.registryClient = registryClient
+        self.fileSystem = fileSystem
+        self.observabilityScope = observabilityScope
+    }
+
+    var indexAndCollections: PackageIndexAndCollections {
+        return .init(fileSystem: fileSystem, observabilityScope: observabilityScope)
+    }
+
+    var repositoryProvider: RepositoryProvider {
+        return GitRepositoryProvider()
+    }
+
+    // FIXME: This matches the current implementation, but we may want be smarter about it?
+    private func guessReadMeURL(baseURL: URL, defaultBranch: String) -> URL {
+        return baseURL.appendingPathComponent("raw").appendingPathComponent(defaultBranch).appendingPathComponent("README.md")
```

Comment:
As mentioned, we might want something better here

Reply:
Yep, and only those which use `README.md` as the file name. I think we should improve on this, but this matches what is being done by the currently shipping Xcode, so seems like a reasonable starting point.